docs/pages/guides/upgrading/0.19.mdx

```diff
@@ -0,0 +1,220 @@
+function argument. You will need to replace instances of `useOther` that do not
+use a selector in your codebase.
+
+#### Previously ❌ [@hidden]
+
+```tsx
+const other = useOther(id); // 👈 does not includes a selector
+```
+
+#### Now ✅ [@hidden]
+
+```tsx
+const other = useOther(id, (other) => other); // 👈 requires a selector
+```
+
+### Stale props support for React 17 [#stale-props]
+
+We've added support to prevent the "zombie-child" scenario. To avoid this issue,
+we enforce passing the `unstable_batchedUpdates` prop to `RoomProvider`.
+
+<Banner title="You may not need this">
+
+This section only applies if you are using React 17 or lower. You do not need to
+do anything if you are using React 18 or higher.
+
+</Banner>
+
+```tsx highlight="2,8"
+// ⚠️ Only if you’re using React 17 or lower
+import { unstable_batchedUpdates }
+from "react-dom"; // 👈
+<RoomProvider
+  id="my-room"
+  initialPresence={...}
+  initialStorage={...}
+  unstable_batchedUpdates={unstable_batchedUpdates}
+>
+  <App />
+</RoomProvider>
+```
+
+For additional context, see the
+[troubleshooting guide](/docs/guides/troubleshooting#stale-props-zombie-child)
+
+### New shouldInitiallyConnect prop [#react-shouldInitiallyConnect]
+
+We added a new property `shouldInitiallyConnect` to `RoomProvider`, which lets
+you control whether or not the room connects to Liveblock servers. By default,
+it will check the `typeof window` to determine if it should connect. When using
+SSR, you can set it to `false` to prevent the room from connecting to Liveblocks
+servers.
+
+```tsx highlight="4"
+<RoomProvider
+  id="room"
+  initialPresence={{}}
+  shouldInitiallyConnect={false} // 👈 control initial connection to Liveblocks
+>
+  <App />
+</RoomProvider>
+```
+
+### Addition of @liveblocks-core package [#liveblocks-core]
+
+We restructured our internal packages to increase code sharing. You may notice a
+new dependency in your dependency tree: `@liveblocks/core`. It contains private
+APIs that aren't intended for direct consumption.
+
+## Changes in @liveblocks/client [#client]
+
+Similar to the [` shouldInitiallyConnect prop`](#react-shouldInitiallyConnect)
+in `@liveblocks/react`, we added a `shouldInitiallyConnect`option
+to`client.enter`. The default value is `true`. You may want to set it to `false`
+when using SSR, or if you would prefer to establish the WebSocket connection
+later.
+
+```tsx
+client.enter("room", {
+  initialPresence: {},
+  shouldInitiallyConnect: false, // 👈 for SSR using the @liveblocks/client package
+});
+```
+
+## Changes in @liveblocks/redux [#redux]
+
+### liveblocksEnhancer is now the default export [#liveblocksEnhancer]
+
+The main export has been renamed, so you will need to update your imports and
+use of the enhancer:
+
+#### Previously ❌ [@hidden]
+
+`import { enhancer } from "@liveblocks/redux";`
+
+#### Now ✅ [@hidden]
+
+`import { liveblocksEnhancer } from "@liveblocks/redux";`
```

Comment:
Why not put it into a code snippet like others? It would benefit from coloring...?